service/history/historyBuilder.go

```diff
@@ -556,18 +557,20 @@ func (b *historyBuilder) newWorkflowTaskFailedEvent(attr *historypb.WorkflowTask
 func (b *historyBuilder) newActivityTaskScheduledEvent(workflowTaskCompletedEventID int64,
 	scheduleAttributes *commandpb.ScheduleActivityTaskCommandAttributes) *historypb.HistoryEvent {
 	historyEvent := b.msBuilder.CreateNewHistoryEvent(enumspb.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED)
-	attributes := &historypb.ActivityTaskScheduledEventAttributes{}
-	attributes.ActivityId = scheduleAttributes.ActivityId
-	attributes.ActivityType = scheduleAttributes.ActivityType
-	attributes.TaskQueue = scheduleAttributes.TaskQueue
-	attributes.Header = scheduleAttributes.Header
-	attributes.Input = scheduleAttributes.Input
-	attributes.ScheduleToCloseTimeout = scheduleAttributes.ScheduleToCloseTimeout
-	attributes.ScheduleToStartTimeout = scheduleAttributes.ScheduleToStartTimeout
-	attributes.StartToCloseTimeout = scheduleAttributes.StartToCloseTimeout
-	attributes.HeartbeatTimeout = scheduleAttributes.HeartbeatTimeout
-	attributes.WorkflowTaskCompletedEventId = workflowTaskCompletedEventID
-	attributes.RetryPolicy = scheduleAttributes.RetryPolicy
+	attributes := &historypb.ActivityTaskScheduledEventAttributes{
+		ActivityId:                   scheduleAttributes.ActivityId,
+		ActivityType:                 scheduleAttributes.ActivityType,
+		TaskQueue:                    scheduleAttributes.TaskQueue,
+		Header:                       scheduleAttributes.Header,
+		Input:                        scheduleAttributes.Input,
+		ScheduleToCloseTimeout:       scheduleAttributes.ScheduleToCloseTimeout,
+		ScheduleToStartTimeout:       scheduleAttributes.ScheduleToStartTimeout,
+		StartToCloseTimeout:          scheduleAttributes.StartToCloseTimeout,
+		HeartbeatTimeout:             scheduleAttributes.HeartbeatTimeout,
+		WorkflowTaskCompletedEventId: workflowTaskCompletedEventID,
+		RetryPolicy:                  scheduleAttributes.RetryPolicy,
+		Namespace:                    scheduleAttributes.Namespace,
```

Comment:
`Namespace` was missed.

Reply:
What is the impact of missing this? Is there any validation we can add?